modules/ephemeris/src/main/scala/gem/horizons/HorizonsEphemerisQuery.scala

```diff
@@ -151,42 +147,45 @@ object HorizonsEphemerisQuery {
     * @return List of queries that cover the entire time range
     */
   def paging(
-        key:   HorizonsDesignation,
-        site:  Site,
-        start: Instant,
-        end:   Instant,
-        step:  Duration): List[HorizonsEphemerisQuery] = {
+    key:   HorizonsDesignation,
+    site:  Site,
+    start: Instant,
+    end:   Instant,
+    step:  Duration
+  ): List[HorizonsEphemerisQuery] = {
+
+    // Calculates the list of queries with adjusted end time and step size.
+    def calc(end: Instant, step: Duration): List[HorizonsEphemerisQuery] = {
+      val maxPageDuration = step * (MaxElements - 1).toLong
+
+      Stream.unfold(start) { s =>               // page start time
```

Comment:
Can you tell a bit more of `Stream.unfold`?

Reply:
It's a way of generating a `Stream` from a seed value and a function.

```
def unfold[S, O](s: S)(f: S => Option[(O, S)]): Stream[Pure, O]
```

The description in the doc is actually pretty good:

> Creates a stream by successively applying `f` until a `None` is returned, emitting each output `O` and using each output `S` as input to the next invocation of `f`.

So in this case `S` is an `Instant` beginning with the start time for the first horizons query.   Each successive invocation gets the start time for that particular page.  We stop, returning `None`, when the page start time has reached or gone past the end.  If we are not stopping though we produce a `Some` containing a tuple `(HorizonsEphemerisQuery, Instant)` where the query gets that page of ephemeris data and the `Instant` is the next page's start time.

The end result is a `Stream[Pure, HorizonsEphemerisQuery]` because the `Instant`s are just used in generating the result.